### PR TITLE
Added onnx export functionality for LpNormNormalizingTransformer

### LINKED ISSUE #4159: LpNormNormalizingTransformer does not support exporting to Onnx
- **What did you do?**
Created a pipeline from NormalizeLpNorm and tried to export it to Onnx with ConvertToOnnxProtobuf. 

- **What happened?**
The transform wasn't saved in the onnx graph

- **What did you expect?**
The transform should be saved in the onnx graph

### PR DESCRIPTION
**Pull request by harishsk**
Fixes #4159

### CODE REVIEW

File: test/Microsoft.ML.Tests/OnnxConversionTest.cs [hunk 1]
```diff
@@ -180,6 +180,68 @@ public void KmeansOnnxConversionTest()
...
+                LpNormNormalizingEstimatorBase.NormFunction.Infinity,
+                LpNormNormalizingEstimatorBase.NormFunction.StandardDeviation
+            };
+
+            bool[] ensureZeroMeans = { true, false};
+            foreach (var ensureZeroMean in ensureZeroMeans)
+            {
+                foreach (var norm in norms)
+                {
+                    var pipe = mlContext.Transforms.NormalizeLpNorm("Features", norm:norm, ensureZeroMean: ensureZeroMean);
```
Comment (codemzs) [inline test/Microsoft.ML.Tests/OnnxConversionTest.cs]: >"Features" [](start = 68, length = 10)

nameof(DataPoint.Features)

File: test/Microsoft.ML.Tests/OnnxConversionTest.cs [hunk 2]
```diff
@@ -180,6 +180,68 @@ public void KmeansOnnxConversionTest()
...
+                    // Compare results produced by ML.NET and ONNX's runtime.
+                    if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows) && Environment.Is64BitProcess)
+                    {
+                        // Evaluate the saved ONNX model using the data used to train the ML.NET pipeline.
+                        string[] inputNames = onnxModel.Graph.Input.Select(valueInfoProto => valueInfoProto.Name).ToArray();
+                        string[] outputNames = onnxModel.Graph.Output.Select(valueInfoProto => valueInfoProto.Name).ToArray();
+                        var onnxEstimator = mlContext.Transforms.ApplyOnnxModel(outputNames, inputNames, onnxModelPath);
+                        var onnxTransformer = onnxEstimator.Fit(dataView);
+                        var onnxResult = onnxTransformer.Transform(dataView);
+                        CompareSelectedR4VectorColumns("Features", "Features1", transformedData, onnxResult, 3);
```
Comment (codemzs) [inline test/Microsoft.ML.Tests/OnnxConversionTest.cs]: >Features [](start = 56, length = 8)

nameof(DataPoint.Features)

Comment (codemzs) [inline test/Microsoft.ML.Tests/OnnxConversionTest.cs]: >Features1 [](start = 68, length = 9)

Can you please get this name from outputNames array?

File: src/Microsoft.ML.Transforms/GcnTransform.cs [hunk 3]
```diff
@@ -594,6 +601,134 @@ private static float Mean(ReadOnlySpan<float> src, int length)
...
+                        strNorm = "L1";
+                    }
+                    else if (_norms[iinfo] == LpNormNormalizingEstimatorBase.NormFunction.L2)
+                    {
+                        strNorm = "L2";
+                    }
+                    else
+                    {
+                        strNorm = "MAX";
+                    }
```
Comment (codemzs) [inline src/Microsoft.ML.Transforms/GcnTransform.cs]: since they are single line we prefer to not add braces.

**Comment by harishsk** [comment]
@codemzs: Pinging for review/approval

File: test/Microsoft.ML.Tests/OnnxConversionTest.cs [hunk 4]
```diff
@@ -180,6 +180,68 @@ public void KmeansOnnxConversionTest()
...
+                    var transformedData = model.Transform(dataView);
+                    var onnxModel = mlContext.Model.ConvertToOnnxProtobuf(model, dataView);
+
+                    var onnxFileName = $"LpNorm-{norm.ToString()}-{ensureZeroMean}.onnx";
+                    var onnxModelPath = GetOutputPath(onnxFileName);
+
+                    SaveOnnxModel(onnxModel, onnxModelPath, null);
+
+                    // Compare results produced by ML.NET and ONNX's runtime.
+                    if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows) && Environment.Is64BitProcess)
```
Comment (ganik) [inline test/Microsoft.ML.Tests/OnnxConversionTest.cs]: >RuntimeInformation.IsOSPlatform(OSPlatform.Windows) [](start = 24, length = 51)

Do you need this condition? If its a linux will results match?

Comment (harishsk) [reply test/Microsoft.ML.Tests/OnnxConversionTest.cs]: I think it is that the test will run only on Windows. The results should still match. It appears that OnnxRuntime doesn't support Linux and Mac yet.

**Pull request merged**